### DISCUSSION TITLE
Add other compnent feature to existed GameObj

### DISCUSSION THREAD

**Discussion by white16819** [kaplayjs/kaplay#510] [Q&A] [answered]
I'm newer to kaplay and ts.
How to add AreaComp's feature such as `onCollide `to a existed GameObj like below:
`const text = k.add([k.text("Test")])`

 I try to use function like this below:
`text.use(k.area())`

Expect the 'text' GameObj's type change into  `GameObj<TextComp | AreaComp>`,
but the 'text' Gameobj's type still be `GameObj<TextComp>`.

So if I invoke the 'onCollide' function in AreaComp, the ts check will raise the error:
`Property 'onCollide' does not exist on type 'GameObj<TextComp>'`

Waiting for your suggestions.

**Comment by lajbel** [comment, marked as answer]
Sure, will add something like

text = text.use(k.area()) return the correct type

**Comment by white16819** [comment]
Thank you, sir
